lua/presenting/init.lua

```diff
@@ -368,4 +384,21 @@ end
 ---@return boolean
 H.in_presenting_mode = function() return Presenting._state ~= nil end
 
+H.extract_title = function(slide)
+  local title = ""
+  for _, line in pairs(vim.split(slide, "\n")) do
+    -- if line is nonempty set it as title
+    if line:match("%S") then
+      title = line
+      break
+    end
+  end
+  -- strip any trailing whitespace
+  title = title:gsub("%s+$", "")
+  -- strip any starting * or # for org and md
+  title = title:gsub("^%*+%s*", "")
+  title = title:gsub("^#+%s*", "")
+  return title
+end
+
```

Comment:
The logic here is not quite apt to the task.

I opened the adoc file and got this title in the footer:
![image](https://github.com/user-attachments/assets/650b8a1b-a1f2-41a2-b1ee-cdc2a7472493)
It works for md and org files. Let's make it work for adoc files as well.

Also I wonder if we can reuse the separator definitions from the config instead of having two very similar definitions.

Reply:
Yeah sure go right ahead! I'm not really using `adoc`.

Ultimately we should add a proper frontmatter parser like in [lookatme](https://lookatme.readthedocs.io/en/latest/) - #12

btw I held 6 classes last week with this and it was awesome - thanks! :pray:

Reply:
Glad to hear that `presenting.nvim` works for you :heart: 

Having a frontmatter parer would be nice, but I want to keep `presenting.nvim` as simple as possible and don't want any external deps.

Reply:
We can do it in plain Lua, e.g. if the first line in the first slide starts with `title:` then we parse the title etc. This is just to populate the footer of course.